### PR TITLE
only print to stdout, if there is a diff

### PR DESCRIPTION
**Pull request by marcb1**
This is for argocd-bot [here](https://github.com/marcb1/argocd-bot). Makes it easier to display diffs on PR's.
When no diff exists for an app, we don't want to comment anything on the PR, so stdout should be empty.

### CODE REVIEW

**Comment by jessesuen** [comment]
@marcb1 do you mind signing the CLA?

**Comment by marcb1** [comment]
@jessesuen sure, done!

File: cmd/argocd/commands/app.go [hunk 1]
```diff
@@ -652,7 +652,7 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 	var command = &cobra.Command{
 		Use:   "diff APPNAME",
 		Short: shortDesc,
-		Long:  shortDesc + "\nUses 'diff' to render the difference. KUBECTL_EXTERNAL_DIFF environment variable can be used to select your own diff tool.",
+		Long:  shortDesc + "\nUses 'diff' to render the difference. KUBECTL_EXTERNAL_DIFF environment variable can be used to select your own diff tool.\nReturns the following exit codes: 1 on error, 2 when a diff is found, 0 when no diff is found",
```
Comment (jessesuen) [inline cmd/argocd/commands/app.go]: Can we swap the exit codes? diff uses the opposite of this (1 for differences, 2 for general errors):

```
$ diff /asdf /dfdfd
diff: /asdf: No such file or directory
diff: /dfdfd: No such file or directory

$ echo $?
2

$ diff /etc/hosts /etc/hosts.equiv
1,9d0
< ##
< # Host Database
< #
< # localhost is used to configure the loopback interface
< # when the system is booting.  Do not change this entry.
< ##
< 127.0.0.1        localhost
< 255.255.255.255  broadcasthost
< ::1              localhost

$ echo $?
1
```

Comment (marcb1) [reply cmd/argocd/commands/app.go]: 👍

File: cmd/argocd/commands/app.go [hunk 2]
```diff
@@ -759,6 +759,7 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 					}
 
 					printDiff(item.key.Name, target, live)
+					os.Exit(1)
```
Comment (jessesuen) [inline cmd/argocd/commands/app.go]: Sorry didn't catch this earlier. Can we exit after we print all the diffs, instead of bailing on the first one?

**Comment by marcb1** [comment]
This failed while building , 
```
unable to deduce repository and source type for "k8s.io/apiextensions-apiserver": unable to
 read metadata: unable to fetch raw metadata: failed HTTP request to URL "http://k8s.io/apiextensions-api
server?go-get=1": Get https://k8s.io/apiextensions-apiserver?go-get=1: net/http: TLS handshake timeout
```

File: cmd/argocd/commands/app.go (hunk 2 above)
Comment (marcb1) [reply cmd/argocd/commands/app.go]: fixed!

**Review by jessesuen — APPROVED** [review]

**Pull request merged**